tck/rest/src/main/java/org/eclipse/microprofile/metrics/test/MicroprofileMetricsTest2.java

```diff
@@ -0,0 +1,158 @@
+
+import static com.jayway.restassured.RestAssured.given;
+
+import com.jayway.restassured.path.json.JsonPath;
+import com.jayway.restassured.response.Header;
+import javax.inject.Inject;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.junit.InSequence;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Heiko W. Rupp
+ */
+@RunWith(Arquillian.class)
+public class MicroprofileMetricsTest2 {
+
+  private static final String APPLICATION_JSON = "application/json";
+
+  @Inject
+  private MetricAppBean2 metricAppBean;
+
+
+  @Deployment
+  public static JavaArchive createDeployment() {
+      JavaArchive jar = ShrinkWrap.create(JavaArchive.class).addClass(MetricAppBean2.class)
+              .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+
+      return jar;
+  }
+
+
+  @Test
+  @InSequence(1)
+  public void setA() {
+    metricAppBean.countMeA();
+    metricAppBean.meterMeA();
+    metricAppBean.timeMeA();
+  }
+
+  @Test
+  @RunAsClient
+  @InSequence(2)
+  public void testSharedCounter() {
+
+    Header acceptJson = new Header("Accept", APPLICATION_JSON);
+
+    JsonPath path = given().header(acceptJson).get("/metrics/application").jsonPath();
+
+    assert path.getInt("countMe2") == 1;
+    assert path.getInt("'org.eclipse.microprofile.metrics.test.MetricAppBean2.meterMe2'.count") == 1;
+    assert path.getInt("timeMe2.count") == 1;
+
+
+  }
+
+  @Test
+  @InSequence(3)
+  public void setB() {
+    metricAppBean.countMeB();
+    metricAppBean.meterMeB();
+    metricAppBean.timeMeB();
+  }
+
+  @Test
+  @RunAsClient
+  @InSequence(4)
+  public void testSharedCounterAgain() {
+
+    Header acceptJson = new Header("Accept", APPLICATION_JSON);
+
+    JsonPath path = given().header(acceptJson).get("/metrics/application").jsonPath();
+
+    assert path.getInt("countMe2") == 2;
+    assert path.getInt("'org.eclipse.microprofile.metrics.test.MetricAppBean2.meterMe2'.count") == 2;
+    assert path.getInt("timeMe2.count") == 2;
+
+  }
+
+  @Test
+  @InSequence(5)
+  public void setReusableHistogram() {
+    metricAppBean.registerReusableHistogram();
+  }
+
+
+  @Test
+  @RunAsClient
+  @InSequence(6)
+  public void testReusedHistogram() {
+
+    Header acceptJson = new Header("Accept", APPLICATION_JSON);
+
+    JsonPath path = given().header(acceptJson).get("/metrics/application").jsonPath();
+
+    assert path.getInt("reusableHisto.count") == 2;
+    assert path.getInt("reusableHisto.min") == 1;
+    assert path.getInt("reusableHisto.max") == 3;
+
+  }
+
+  @Test
+  @InSequence(7)
+  public void testBadReusable() {
+
+    try {
+      metricAppBean.badRegisterReusableHistogram();
+    }
+    catch (Exception e) {
```

Comment:
should this catch RuntimeException rather than Exception?  I'm suggesting RuntimeException because throwing a checked exception would result in callers needing to catch or rethrow the exception that could occur during registration.  since the caller can't ever correct for this situation (it really just indicates a code bug) it seems that an unchecked exception would be better.

Reply:
Not sure I get this one.
This is in context of testing and works for IAE - perhaps we should explicitly test for IAE when the spec says to throw IAE in such cases?

Reply:
Right, I was meaning we should explicitly catch IAE (and only IAE) to ensure that's what the impl throws in this case.  (assuming you agree IAE is good/right exception to throw).